// @ionic/cli-utils/lib/environment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Debug = require("debug");
const cli_framework_1 = require("@ionic/cli-framework");
const logger_1 = require("./utils/logger");
const debug = Debug('ionic:cli-utils:lib:environment');
class Environment {
    constructor({ client, config, flags, getInfo, log, ctx, prompt, session, shell, tasks }) {
        this.client = client;
        this.config = config;
        this.flags = flags;
        this.getInfo = getInfo;
        this.log = log;
        this.ctx = ctx;
        this.prompt = prompt;
        this.session = session;
        this.shell = shell;
        this.tasks = tasks;
    }
    open() {
        if (this.flags.interactive) {
            this.prompt.open();
        }
        const formatter = logger_1.createFormatter();
        this.log.handlers = this.flags.interactive
            ? new Set([new cli_framework_1.StreamHandler({ stream: this.prompt.output.stream, formatter })])
            : new Set([...cli_framework_1.DEFAULT_LOGGER_HANDLERS].map(handler => handler.clone({ formatter })));
        debug('Environment open.');
    }
    close() {
        this.tasks.cleanup();
        this.prompt.close();
        const formatter = logger_1.createFormatter();
        this.log.handlers = new Set([...cli_framework_1.DEFAULT_LOGGER_HANDLERS].map(handler => handler.clone({ formatter })));
        debug('Environment closed.');
    }
}
exports.Environment = Environment;
