// ionic/dist/commands/capacitor/copy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const chalk_1 = require("chalk");
const base_1 = require("./base");
class CopyCommand extends base_1.CapacitorCommand {
    getMetadata() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return {
                name: 'copy',
                type: 'project',
                summary: 'Copies web assets to each Capacitor native platform',
                description: `
${chalk_1.default.green('ionic capacitor copy')} will do the following:
- Copy the ${chalk_1.default.bold('www/')} directory into your native platforms.
      `,
                exampleCommands: [],
                inputs: [],
            };
        });
    }
    preRun(inputs, options, runinfo) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            yield this.preRunChecks(runinfo);
        });
    }
    run(inputs, options) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            yield this.runCapacitor(['copy']);
        });
    }
}
exports.CopyCommand = CopyCommand;
