// ionic/dist/commands/capacitor/open.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const chalk_1 = require("chalk");
const base_1 = require("./base");
class OpenCommand extends base_1.CapacitorCommand {
    getMetadata() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            return {
                name: 'open',
                type: 'project',
                summary: 'Opens the IDE for the specified Capacitor native platform project',
                description: `
${chalk_1.default.green('ionic capacitor open')} will do the following:
- Open the IDE with your current native project for that specific platform. This means Xcode for iOS, and Android Studio for Android
      `,
                exampleCommands: [],
                inputs: [],
            };
        });
    }
    preRun(inputs, options, runinfo) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            yield this.preRunChecks(runinfo);
        });
    }
    run(inputs, options) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const [platform] = inputs;
            const args = ['open'];
            if (platform) {
                args.push(platform);
            }
            yield this.runCapacitor(args);
        });
    }
}
exports.OpenCommand = OpenCommand;
